tests/Server/TextDocument/SignatureHelpTest.php

```diff
@@ -0,0 +1,210 @@
+                    'activeSignature' => 0,
+                    'activeParameter' => 1,
+                ]),
+            ],
+            'method with no params' => [
+                new Position(51, 9),
+                $this->createSignatureHelp([
+                    'label' => '()',
+                    'documentation' => 'Method with no params',
+                    'parameters' => [],
+                    'activeSignature' => 0,
+                    'activeParameter' => 0,
+                ]),
+            ],
+            'constructor' => [
+                new Position(47, 14),
+                $this->createSignatureHelp([
+                    'label' => '(string $first, int $second, \Foo\Test $third)',
+                    'documentation' => 'Constructor comment goes here',
+                    'parameters' => [
+                        [
+                            'label' => 'string $first',
+                            'documentation' => 'First param',
+
+                        ],
+                        [
+                            'label' => 'int $second',
+                            'documentation' => 'Second param',
+                        ],
+                        [
+                            'label' => '\Foo\Test $third',
+                            'documentation' => 'Third param with a longer description',
+                        ],
+                    ],
+                    'activeSignature' => 0,
+                    'activeParameter' => 0,
+                ]),
+            ],
+            'global function' => [
+                new Position(53, 4),
+                $this->createSignatureHelp([
+                    'label' => '(int $i, bool $b = false)',
+                    'documentation' => null,
+                    'parameters' => [
+                        [
+                            'label' => 'int $i',
+                            'documentation' => 'Global function param one',
+                        ],
+                        [
+                            'label' => 'bool $b = false',
+                            'documentation' => 'Default false param',
+                        ],
+                    ],
+                    'activeSignature' => 0,
+                    'activeParameter' => 0,
+                ]),
+            ],
+            'static method' => [
+                new Position(55, 10),
+                $this->createSignatureHelp([
+                    'label' => '(mixed $a)',
+                    'documentation' => null,
+                    'parameters' => [
+                        [
+                            'label' => 'mixed $a',
+                            'documentation' => null,
+                        ],
+                    ],
+                    'activeSignature' => 0,
+                    'activeParameter' => 0,
+                ]),
+            ],
+        ];
+    }
+
+    private function createSignatureHelp(array $info): SignatureHelp
```

Comment:
I've added constructors and utilised them (I actually found this style of OO with public members a bit strange but just copied the existing style). I still like to have this creator as otherwise we end up constructing lots of objects in the data providers and it becomes harder to maintain - if a constructor changed there would be a lot of rework.

Reply:
Please use the constructors directly - especially after your PR, you can get signature help for the constructor parameters, but you can't for the array hashes ;)
That's how it's done in all the other tests too.

Reply:
There is no reason to make the members private and write getters and setters for them as they represent the JSON that gets sent over the wire (which only has public properties). They are generally dumb value objects that don't contain any complex state that needs to be kept private.

Reply:
Got it, makes sense - updated now, thanks